.github/workflows/CI.yml

```diff
@@ -21,7 +21,7 @@ jobs:
     needs: Get-CI-Image-Tag
     strategy:
       matrix:
-        java: [21, 23]
+        java: [21, 24]
```

Comment:
why we dropping version 23? It's not like we moving to latest, we still keep version 21.

Reply:
ignore my question, I see this was the ask from infra team

Reply:
Yes @martin-gaievski.

JDK23 is not LTS and JDK21 is base. The idea is while we keep 21 as min compatible version, we will keep testing the latest JDK version to ensure its functionality and support. As this has always been the case following core updating JDK.

Thanks.